opts.go

```diff
@@ -271,3 +314,30 @@ func loadFromConfDir(c *libcni, max int) error {
 	c.networks = append(c.networks, networks...)
 	return nil
 }
+
+func checkPluginExists(c *libcni, confList *cnilibrary.NetworkConfigList) error {
+	var err error
+	for _, plug := range confList.Plugins {
+		plugin := plug.Network.Type
+		for _, dir := range c.pluginDirs {
+			if !fileExistsInDir(dir, plugin) {
+				err = fmt.Errorf("plugin '%s' not found in: %s %v", plugin, strings.Join(c.pluginDirs, ","), ErrCNIPluginNotFound)
+			} else {
+				err = nil
+				continue
+			}
+		}
+
+		if err != nil {
+			return err
```

Comment:
nod looks good or.. and as per offline discussion could append the missing plugins into a list and output them all in the err..